Release 1.19 (#313)
cdr/docs: changelog/1.19.0.md

---
title: "1.19.0"
description: "Released on 05/19/2021"
---

### Breaking changes ❗

There are no breaking changes in 1.19.0.

### Features ✨

- web: Resource quotas have been introduced to Organizations. Resource Quotas define
  maximum resource utilization applied to each member within the Organization they're

>   applied to. Modify them **Manage > Organizations > Edit Organization**.

Author: 1. Need to fix up grammar
2. Needs documentation

> - web: Add site and organization workspace as code templates

Author: 1. Needs documentation

Reviewer: Is this referring to the nested templates that are now pushed back, or is this something else @Emyrk?

Author: I update this clog periodically during the dev cycle. I haven't made an update since we cut, so this will be removed once I do that.

Reviewer 2: That is

> - web: An administrative control over Dev URL access levels was added.

Author: Needs documentation

> - web: By default, the CVM option for creating custom environments is now checked

Author: Needs documentation

Reviewer: Does this mean that CVMs are enabled at the admin level (Admin > Infrastructure> Workspace Container Runtime --> Enable Container-Based Virtual Machines) automatically, too?

Author: Good point, I'll follow up. I don't think so, but need to confirm. As far as I understand, it's just that when that setting is checked, CVM is checked when creating a workspace using the UI by defauktl

- web: Improved display and support for errors returned by an IdP during sign in

> - infra: Use stdio for envassets

Author: Need help writing this changelog (unsure how to word) cc @kylecarbs

Author: Thus I think this should:

a) move to Breaking Changes
b) Be reworded to somethign along the lines of:

```
infra: Workspace assets previously located in the filepath /opt/coder, such as code-server and coder-cli, have now been moved to /var/tmp/coder
```

Reviewer 3: This is a breaking change, as some of our customers use that code-server binary to install extensions on their workspace.

We've also mentioned this location a few times in our docs, including the "Images" docs: https://github.com/cdr/docs/search?q=%2Fopt%2Fcoder

Author: ^ @bpmct :confetti_ball:

### Bug fixes 🐛

- web: Fixed an issue whereby in the **Edit Workspace** dialog, resource selectors
  displayed as disabled but still allowed modification
- web: Updated validation logic so that workspaces with invalid names can no longer
  be created
- infra: Fixed an issue that caused workspace autostart to sometimes halt or fail

### Security updates 🔐

> - web: Upgrade version of NextJS to 10.2.0

Reviewer 4: perhaps we should do a scan of 1.18.1 and the latest version and list out all the fixed CVEs?

Author: I'm good w that - will need help of @mterhar

  for which such quotas are enabled. Modify them at **Manage** >
  **Organizations** > **Edit Organization**.
- web: Added site and organization workspace as code templates.
- web: Added administrative control over dev URL access levels.
- web: The CVM option when creating a custom environment is now selected by default.
- web: Improved display and support for errors returned by an IdP during sign-in
- infra: Changed permissions of `/home/coder/` from _755_ to _744_
- infra: Workspace builds are now validated when setting policy templates
- infra: Added client certificate authentication support

### Bug fixes 🐛

- web: Fixed issue where the **Edit Workspace** dialog showed the resource
  selectors as disabled while still allowing modifications
- web: Fixed issue where the **Edit Workspace** dialog displayed resource
  allocation incorrectly if the workspace was initially built using a template
- web: Updated validation logic so that workspaces with invalid names can no
  longer be created
- infra: Fixed issue causing workspace autostart to stop or fail intermittently
- web: Fixed issue in the **Audit Logs** preventing users from clearing the
  filters

> ### Security updates 🔐

Reviewer 4: I guess we should add a section here about updates to certificate handling support, particularly that (a) we now have debug logs for certificates that we load, and (b) we can reload certificates with a SIGHUP is received. I'm not sure whether these belong under security or features, maybe features.

Maybe: "infra: Add debug logs showing TLS certificates, and reload certificates when a SIGHUP signal is received" ?

---
title: "1.19.0"
description: "Released on 05/19/2021"
---

> ### Breaking changes ❗

Reviewer 5: I believe WAC was updated to `0.2` and therefore `0.1` templates no longer work. We may want to call this out. /cc @Emyrk @kylecarbs

Author: It needs product documentation too.

Reviewer 2: I've been having issues rebasing/merging master into my PR:https://github.com/cdr/docs/pull/304

Given we are pushing nested templates back a bit, I might trim the `policy` fields for now

Reviewer 2: See text on: https://codercom-67d5emquz-codercom.vercel.app/docs/workspaces/workspaces-as-code/templates

Is that sufficient?

Reviewer: @Emyrk I'll take a look at that PR

Reviewer 2: Thanks @khorne3. I was having some issues rebasing and getting the pre-commit stuff to pass. When I merged master in, and couldn't pass the pre-hook, I was unable to commit my merge :smile:

- infra: Workspaces as code templates have been updated from version _0.1_ to
  _0.2_. Version _0.1_ templates will no longer work.
- web: The flow for workspaces as code no longer includes a clone step. The form
  to build a button located at **Manage** > **Admin** > **Templates** no longer
  uses the following fields: _Project Repository URL_ and _Project Git Service_.
  Buttons created from prior versions will still work, but a project will not be
  cloned as part of the flow. In order to automate project cloning, use
  <https://coder.com/docs/workspaces/workspaces-as-code/templates#workspaceconfigurestart>.

### Features ✨

- web: Introduced resource quotas to organizations. Resource quotas define the
  maximum resource utilization allowable for each member within an organization
  for which such quotas are enabled. Modify them at **Manage** >
  **Organizations** > **Edit Organization**.
- web: Added a reactivation step for returning users that had been marked as
  dormant. The reactivation step require the user to agree to consume a licensed
  seat.
- web: Added administrative control over dev URL access levels

> - infra: Added client certificate authentication support

Reviewer 4: let's skip this please :)

Author: Done.

- web: The CVM option when creating a custom environment is now selected by

>   default when its enabled at the site level.

Reviewer 4: ```suggestion
  default when it is enabled at the site level.
```

> - web: Improved display and support for errors returned by an IdP during sign-in

Reviewer 4: ```suggestion
- web: Improved display and support for errors returned by an Identity Provider (IdP) during sign-in
```

- infra: Changed permissions of `/home/coder/` from _755_ to _744_

> - infra: Workspace builds are now validated when setting policy templates

Reviewer 6: ```suggestion
- infra: Workspace builds are now properly validated when setting policy templates
```

> - code-server: Includes latest version 3.10.0 of code-server

Reviewer 6: ```suggestion
- code-server: Upgrade code-server to 3.10.0 (latest version)
```

### Bug fixes 🐛

- web: Fixed issue where the **Edit Workspace** dialog showed the resource
  selectors as disabled while still allowing modifications
- web: Fixed issue where the **Edit Workspace** dialog displayed resource
  allocation incorrectly if the workspace was initially built using a template
- web: Updated validation logic so that workspaces with invalid names can no
  longer be created web: Fixed an issue causing web-based Terminals to have a
  double scrollbar and content cut-off.
- infra: Fixed issue causing workspace autostart to stop or fail intermittently
- web: Fixed issue in the **Audit Logs** preventing users from clearing the

>   filters

Reviewer 6: Fixed **an** issue

- web: A _.csv_ extension is now added to exported audit logs

### Security updates 🔐

- web: Upgraded version of Next.js to 10.2.0.

> - web: Upgraded and addressed CVE-2021-21306 in marked.

Reviewer 6: ```suggestion
- web: Upgraded and addressed CVE-2021-21306 in `marked` dependency.
```

etc.

- web: Upgraded and addressed Fix CVE-2021-23368 in postcss.
- web: client secret is now omitted from GET requests to OAuth configurations.

> - infra: Improved use of prepared SQL statements and added tests to ensure the
>   application is guarded from SQL injection.

Reviewer 6: ```suggestion
```

- web: When authenticating with the CLI, a new API key is always generated.
- web: Fixed authentication for the _watch-update_ workspace endpoint. This was
  previously open to any authenticated user
- web: Fixed issue where audit logs for **workspace auto-off** was always
  attributed to the site admin; the audit log has been corrected to display the
  user owning the workspace.

> - infra: Added client certificate authentication support. Client certificates

Reviewer 4: would like to avoid mentioning this for 1.19, we'll mention it in the next release and I'll write some docs for it

Author: Done.

  are laoded from `/etc/ssl/private`
- infra: Increased the minimum requirement for inbound and outbound connections
  to TLS 1.2
- infra: Added debug logs to show TLS certificates

> - infra: Coder automatically reloads certificates when it receives a _SIGHUP_

Reviewer 4: let's skip this too for now, I'm not 100% sure it works correctly and there's some improvements I want to make based on what I learned from @lilshoff

Author: Done.

  URL Access Permissions**
- web: The CVM option will be selected by default when creating a custom
  environment (if CVMs are enabled at the site level)
- web: Improved display and support for errors returned by an Identity Provider
  (IdP) during sign-in
- infra: Changed permissions of `/home/coder/` from _755_ to _744_
- infra: Workspace builds are now properly validated when setting policy
  templates
- code-server: Upgraded code-server to 3.10.0 (latest version)

### Bug fixes 🐛

- web: Fixed issue where the **Edit Workspace** dialog showed the resource
  selectors as disabled while still allowing modifications
- web: Fixed issue where the **Edit Workspace** dialog displayed resource
  allocation incorrectly if the workspace was initially built using a template
- web: Updated validation logic so that workspaces with invalid names can no
  longer be created
- web: Fixed issue causing web-based terminals to show two scrollbars and
  overflowing content when system banners are present
- infra: Fixed issue causing workspace autostart to stop or fail intermittently
- web: Fixed issue in the **Audit Logs** preventing users from clearing the
  filters

> - web: Added a _.csv_ extension to exported audit logs

Author: @khorne3 wondering if these should all be `Fixed an issue` instead of `Fixed issue`.

Reviewer: I think I prefer the terseness of "Fixed issue"?

Author: Sounds good, I defer to you! Was a gut check, wasn't sure.

---
title: "1.19.0"
description: "Released on 05/19/2021"
---

### Breaking changes ❗

- infra: Workspace assets created by Coder that were previously located at
  `/opt/coder`, such as code-server and coder-cli, have been moved to
  `/var/tmp/coder`
- infra: Workspaces as code templates have been updated from version _0.1_ to

>   _0.2_. Version _0.1_ templates will no longer work.

Reviewer 7: Version 0.1 templates will work though... right?

They just can't be updated before they're migrated I thought

Reviewer: I think they *don't* -- @Emyrk can you please confirm?

Reviewer: Confirmed that 0.1 templates don't work beginning w/ v1.19

Reviewer 2: @kylecarbs All 0.1 templates currently in use will continue to work (like you can rebuild your workspace). You cannot upload a new 0.1 template.

### Features ✨

- web: Introduced resource quotas to organizations. Resource quotas define the
  maximum resource utilization allowable for each member within an organization
  for which such quotas are enabled. Modify them at **Manage** >
  **Organizations** > **Edit Organization**
- web: Added a reactivation step for returning users that had been marked as
  dormant. The reactivation step requires the user to agree to use a license
  seat
- web: Added an admin setting controlling the maximum dev URL access level
  across all workspaces within the deployment. Modify the maximum dev URL access
  level at **Manage** > **Admin** > **Infrastructure** in the card titled **Dev
  URL Access Permissions**
- web: The CVM option will be selected by default when creating a custom
  environment (if CVMs are enabled at the site level)
- web: Improved display and support for errors returned by an Identity Provider
  (IdP) during sign-in
- infra: Changed permissions of `/home/coder/` and subdirectories to _755_ in
  control plane containers to support OpenShift `anyuid`
- infra: Workspace builds are now properly validated when setting policy
  templates

> - code-server: Upgraded code-server to 3.10.0 (latest version)

Reviewer 7: 3.10.1 I believe

### Breaking changes ❗

- infra: Workspace assets created by Coder previously located at `/opt/coder`,
  such as code-server and coder-cli, have been moved to `/var/tmp/coder`

### Features ✨

> - web: Introduced resource quotes to organizations. Resource quotas define the

Reviewer 4: ```suggestion
- web: Introduced resource quotas to organizations. Resource quotas define the
```

  maximum resource utilization allowable for each member within an organization
  for which such quotas are enabled. Modify them at **Manage** >
  **Organizations** > **Edit Organization**.
- web: Added site and organization workspace as code templates.
- web: Added administrative control over dev URL access levels.
- web: The CVM option when creating a custom environment is now selected by default.
- web: Improved display and support for errors returned by an IdP during sign-in

> - infra: Changed permissions of `/home/coder/` from _755_ to _744_

Author: I'll find out - good call out

> - infra: Changed permissions of `/home/coder/` from _755_ to _744_

Author: (I'd assume only in workspace containers?)

> - infra: Changed permissions of `/home/coder/` from _755_ to _744_

Reviewer 4: I think the change that @mterhar was to change /home/coder itself from 700 to 755, so that when we run with anyuid on OpenShift, the random uid can execute our binary. This was only in the control plane containers.

> - infra: Changed permissions of `/home/coder/` from _755_ to _744_

Author: It actually affects both the coder-service and dashboard containers
